### PR TITLE
SearchField: Add label option and improve accessibility

### PR DESCRIPTION
**Pull request by ayeshakmaz**
## Description
- Add option to specify a visible label,
- Add label for the clear button
- Make clear button keyboard accessible

### Links

- JIRA: https://jira.pinadmin.com/browse/PDS-2454

### Checklist

- [ ] Added Unit and Flow Tests
- [X] Added documentation + accessibility tests
- [X] Verified Accessibility: keyboard & screen reader interaction
- [X] Checked Dark Mode, Responsiveness, and Right-to-Left support
- [ ] Checked Stakeholder feedback (ex. Gestalt designers)

### CODE REVIEW

File: packages/gestalt/src/SearchField.js [hunk 1]
```diff
@@ -10,12 +10,15 @@ import formElement from './FormElement.css';
 import Box from './Box.js';
 import Icon from './Icon.js';
 import FormErrorMessage from './FormErrorMessage.js';
+import FormLabel from './FormLabel.js';
 import { type AbstractEventHandler } from './AbstractEventHandler.js';
 
 type Props = {|
   accessibilityLabel: string,
+  accessibilityClearButtonLabel?: string,
```
Comment (ayeshakmaz) [inline packages/gestalt/src/SearchField.js:18]: I made this optional for now, but can go back later and make it required and do a codemod

Comment (ayeshakmaz) [reply packages/gestalt/src/SearchField.js:18]: Might make sense to do at the same time as Typeahead

File: docs/src/SearchField.doc.js [hunk 2]
```diff
@@ -2,13 +2,55 @@
...
+card(
+  <PageHeader
+    name="SearchField"
+    defaultCode={`
+function SearchFieldExample() {
+  const [value, setValue] = React.useState('');
+
+  return (
+    <Box padding={3} display="flex" alignItems="center" flex="grow">
+      <Box padding={3}>
```
Comment (dangerismycat) [inline docs/src/SearchField.doc.js:20]: nit: Flex could make this markup cleaner, no need for most of these wrapping Boxes

File: docs/src/SearchField.doc.js [hunk 3]
```diff
@@ -82,22 +130,23 @@ card(
...
+    >
+      <MainSection.Card
+        cardSize="lg"
+        defaultCode={`
   function SearchFieldExample() {
     const [value, setValue] = React.useState('');
 
     return (
-      <Box color="white" rounding={2} padding={3} display="flex" alignItems="center">
+      <Box padding={3} display="flex" alignItems="center" flex="grow">
```
Comment (dangerismycat) [inline docs/src/SearchField.doc.js:149]: same nit here

File: packages/gestalt/src/SearchField.js [hunk 4]
```diff
@@ -157,7 +168,7 @@ const SearchFieldWithForwardRef: React$AbstractComponent<Props, HTMLInputElement
         />
 
         {hasValue && (
-          <button className={styles.clear} onClick={handleClear} tabIndex={-1} type="button">
```
Comment (dangerismycat) [inline packages/gestalt/src/SearchField.js:160]: 😬

**Review by dangerismycat — APPROVED** [review]

**Pull request merged**